src/pause/140C70.c

```diff
@@ -1,17 +1,287 @@
+void pause_map_draw_title(s32* arg1, s32 arg2, s32 textOffsetY, s32 textOffsetX) {
+    s32 temp_v1;
+    s32 stringWidth;
+    s32 stringID;
+    s32 xPos;
+    s32 yPos;
+
+    if (gPauseMenuCurrentTab == 6) {
+        temp_v1 = mapCursorCurrentOption;
+        if (temp_v1 != -1) {
+            stringWidth = get_string_width((temp_v1 * 3) + pause_map_tab_places_description_string_id, 0);
+            stringID = (mapCursorCurrentOption * 3) + pause_map_tab_places_description_string_id;
+            xPos = arg2 + ((textOffsetX - stringWidth) >> 1);
+            yPos = textOffsetY + 1;
+            draw_string(stringID, xPos, yPos, 0xFF, 0, 0);
+        }
+    }
+}
+
+#ifdef NON_MATCHING
+void pause_map_init(s8* arg0) {
+    f32 temp_f0;
+    f32 temp_f0_2;
+    f32 temp_f2;
+    f32 temp_f2_2;
+    f32 temp_f6;
+    s32 i;
+    s32* phi_s2;
+    HudElement* hudElement;
+    s32 phi_s0_2;
+    s32 phi_v0_2;
+    s32 phi_s0_4 = 0;
+    pause_map_spaces* map_spaces;
+    pause_80250668* phi_v0;
+    s32* hudElementPtr = &D_80270700;
+    s32* phi_s1 = &D_8024FA30;
+    
+    
+    for (i = 0; i <= 0; i++) {
+        hudElementPtr[i] = create_hud_element(phi_s1[0]);
+        set_hud_element_flags(hudElementPtr[i], 0x80);
+        phi_s1++;
+    }
+    
+    i = 6;
+    phi_v0 = &D_80250668;
+    
+
+    for (i; i >= 0; i--) {
+        phi_v0->unk_10 = arg0;
+        phi_v0--;
+    }
+
+    setup_pause_menu_tab(&D_80250590, 7);
+    mapCursorCurrentOption = -1;
+    D_80270724 = 0;
+    mapCursorCurrentOptionCopy = -1;
 
+    for (i = 0; i < EVT_SAVE_FLAG_PLACES_VISITED_TOTAL; i++) {
+        if (map_spaces->tabIndex != evt_get_variable(0, 0xF5DE0329)) {
+            map_spaces++;
+            phi_s0_4 = 1;
+        } else {
+            break;
+        }
+    }
+
+    if (map_spaces->tabIndex != 0) {
+        map_spaces = phi_s0_4 * sizeof(pause_map_spaces) + &D_802502B8;
+        mapMarioXPos = map_spaces->xPos;
+        mapMarioYPos = map_spaces->yPos;
+    } else {
+        mapMarioXPos = 0;
+        mapMarioYPos = 0;
+    }
+    
+    temp_f2 = mapMarioXPos;
+    mapCursorXPos = temp_f2;
+    D_80270704 = 0.0f;
+    temp_f6 = D_80270704;
+    temp_f0 = mapMarioYPos;
+    mapCursorYPos = temp_f0;
+    temp_f2_2 = temp_f6 - (f32) (s32) ((f64) (temp_f2 + temp_f6) - D_80270080);
+    temp_f0_2 = temp_f6 - (f32) (s32) ((f64) (temp_f0 + temp_f6) - D_80270088);
+    D_80270708 = 0.0f;
+    D_80270704 = temp_f2_2;
+    D_80270708 = temp_f0_2;
+    if (temp_f6 < temp_f2_2) {
+        D_80270704 = temp_f6;
+    }
+    if (temp_f6 < temp_f0_2) {
+        D_80270708 = temp_f6;
+    }
+    if (D_80270704 <= -86.0f) {
+        D_80270704 = -85.0f;
+    }
+    if (D_80270708 <= -210.0f) {
+        D_80270708 = -209.0f;
+    }
+    *arg0 = 1;
+}
+#else
 INCLUDE_ASM(s32, "pause/140C70", pause_map_init);
+#endif
+
+void pause_map_handle_input(void) {
+    f64 temp_f4_2;
+    s32 temp_f12;
+    pause_map_spaces* map_spaces;
+    GameStatus* gameStatus;
```

Comment:
Imo the very first line of the function should be the GameStatus one and you can initialize it on the same line. then you can initialize those temps right after